surface/include/pcl/surface/mls.h

```diff
@@ -283,6 +306,22 @@ namespace pcl
       inline int
       getDilationIterations () { return dilation_iteration_num_; }
 
+      /** \brief Set wether the mls results should be stored for each point in the input cloud
+        * \param[in] True if the mls results should be stored, otherwise false.
+        */
+      inline void
+      setCacheMLSResults (bool cache_mls_results) { cache_mls_results_ = cache_mls_results; }
```

Comment:
I think we need a warning here in the comments stating that `cache_mls_results_` will be forced to true if the methods used are VOXEL_GRID_DILATION or DISTINCT_CLOUD.